src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameLexer.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame
+{
+    using StackFrameToken = EmbeddedSyntaxToken<StackFrameKind>;
+    using StackFrameTrivia = EmbeddedSyntaxTrivia<StackFrameKind>;
+
+    internal struct StackFrameLexer
+    {
+        public readonly VirtualCharSequence Text;
+        public int Position;
+
+        public StackFrameLexer(string text)
+            : this(VirtualCharSequence.Create(0, text))
+        {
+        }
+
+        public StackFrameLexer(VirtualCharSequence text) : this()
```

Comment:
I assume `VirtualCharSequence` is for expediency or do we actually get useful features from it? I would expect that we just care about strings

Reply:
Not using any of the features right now, but this sets us up really nicely for if we want to add this as an embedded language somehow in the future. It's not much harder than using a string directly and callers don't really need to worry about it.

Reply:
at first i didn't see the point of this.  but being able to read these in a string literal regardless of the characters used does seem nice.